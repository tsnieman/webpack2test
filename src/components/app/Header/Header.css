@import 'styles/_variables.css';

.wrapper {
  background: var(--color-brand-primary);
  padding: 1rem;
  text-align: center;
}

.brand {
  font-size: 2em;

  a {
    color: var(--color-white);
  }
}

.nav-items {
  margin: 1em 0 0 0;
  padding: 0;

  a {
    color: var(--color-white);
  }

  & > * {
    display: inline-block;
    vertical-align: middle;
    padding: 0.1em 0.5em;
    border-bottom: 2px solid color(var(--color-white) alpha(25%));
    /*background: color(var(--color-white) alpha(25%));*/

    + * {
      margin-left: 1rem;
    }
  }
}
