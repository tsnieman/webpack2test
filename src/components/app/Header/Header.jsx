import React from 'react';
import { Link } from 'react-router-dom';

const Header = () => (
  <header>
    <strong><Link to="/">🍞 webpack2test</Link></strong>

    <nav>
      <ul>
        <li>
          <Link to="/about">/about</Link>
        </li>

        <li>
          <Link to="/404-fake-route">/404-fake-route</Link>
        </li>
      </ul>
    </nav>
  </header>
);

Header.propTypes = {
};

export default Header;
