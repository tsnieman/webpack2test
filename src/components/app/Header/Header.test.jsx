import React from 'react';
import renderer from 'react-test-renderer'; // eslint-disable-line import/no-extraneous-dependencies
import { MemoryRouter } from 'react-router-dom';
import Header from './Header';

test('Header is tested', () => {
  const component = renderer.create(
    <MemoryRouter>
      <Header />
    </MemoryRouter>,
  );

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
