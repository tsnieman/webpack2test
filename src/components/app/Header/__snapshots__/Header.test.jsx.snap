// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header is tested 1`] = `
<header>
  <strong>
    <a
      href="/"
      onClick={[Function]}
    >
      🍞 webpack2test
    </a>
  </strong>
  <nav>
    <ul>
      <li>
        <a
          href="/about"
          onClick={[Function]}
        >
          /about
        </a>
      </li>
      <li>
        <a
          href="/404-fake-route"
          onClick={[Function]}
        >
          /404-fake-route
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
